trajopt/src/problem_description.cpp

```diff
@@ -589,47 +625,104 @@ void JointVelTermInfo::fromJson(ProblemConstructionInfo& pci, const Json::Value&
   FAIL_IF_FALSE(v.isMember("params"));
   const Json::Value& params = v["params"];
 
+  unsigned int n_dof = pci.kin->numJoints();
   json_marshal::childFromJson(params, coeffs, "coeffs");
+
+  // Optional Parameters
+  json_marshal::childFromJson(params, targs, "targs", DblVec(n_dof, 0));
+  json_marshal::childFromJson(params, upper_tols, "upper_tols", DblVec(n_dof, 0));
+  json_marshal::childFromJson(params, lower_tols, "lower_tols", DblVec(n_dof, 0));
   json_marshal::childFromJson(params, first_step, "first_step", 0);
   json_marshal::childFromJson(params, last_step, "last_step", pci.basic_info.n_steps - 1);
-  unsigned n_dof = pci.kin->numJoints();
-  if (coeffs.size() == 1)
-  {
-    coeffs = DblVec(n_dof, coeffs[0]);
-    ROS_INFO("1 JointVelTermInfo coefficient given. Applying to all %i joints", n_dof);
-  }
-  else if (coeffs.size() != n_dof)
-  {
-    PRINT_AND_THROW(boost::format("wrong number of coeffs. expected %i got %i") % n_dof % coeffs.size());
-  }
 
-  const char* all_fields[] = { "coeffs", "first_step", "last_step" };
+  const char* all_fields[] = { "coeffs", "first_step", "last_step", "targs", "lower_tols", "upper_tols" };
```

Comment:
Is `targs` the best name? I can't think of anything better at the moment but thought I would ask.

Reply:
I asked a few people around the office. Candidates that were mentioned were goal, targets, or nominal. Goal seems to make sense, the only caveat being that it (and target as well) seems to imply that we are moving toward that point. That might not be the case for say acceleration limits. However, nominal seems unintuitive.